src/Compilers/CSharp/Portable/Binder/DecisionDagBuilder.cs

```diff
@@ -1258,6 +1307,13 @@ private static ImmutableArray<StateForCase> RemoveEvaluation(ImmutableArray<Stat
                     }
                     break;
             }
+
+            static int getCanonicalIndex(BoundDagIndexerEvaluation e)
+            {
+                return e.Index;
+                // TODO Adjust sliced indexer evaluations
```

Comment:
:memo: This won't work because an inner list wouldn't be a successor to the outer pattern in a multi-arm match. If we don't want this to affect codegen, we may need to maintain a separate dag only for subsumption purposes.